infra/scripts/test-end-to-end-batch-dataflow.sh

```diff
@@ -4,178 +4,249 @@ echo "Preparing environment variables..."
 
-1. Install gcloud SDK and required packages.
-2. Create temporary BQ table for Feast Serving.
-3. Generate valid names for IP addresses and k8s cluster, then store as environment variables.
-4. Create GKE nodepool for Feast e2e test with DataflowRunner.
-5. Setup Feast Core, Feast Serving and dependencies using Helm.
-  - Redis as the job store for Feast Batch Serving.
-  - Postgres for persisting Feast metadata.
-  - Kafka and Zookeeper as the Source in Feast.
-6. Install Python 3.7.4, Feast Python SDK and run end-to-end tests from
+1. Setup K8s cluster (optional, if it was not created before)
+2. Reuse existing IP addresses or generate new ones for stateful services
+3. Install stateful services (kafka, redis, postgres, etc) (optional)
+4. Build core & serving docker images (optional)
+5. Create temporary BQ table for Feast Serving.
+6. Rollout target images to cluster via helm in dedicated namespace (pr-{number})
+7. Install Python 3.7.4, Feast Python SDK and run end-to-end tests from
    tests/e2e via pytest.
-7. Tear down infrastructure, including Dataflow jobs.
+8. Tear down feast services, keep stateful services.
 "
 
 ORIGINAL_DIR=$(pwd)
 echo $ORIGINAL_DIR
 
-echo "
-============================================================
-Installing gcloud SDK and required packages
-============================================================
-"
-apt-get -qq update
-apt-get -y install wget netcat kafkacat build-essential gettext-base curl kubectl
-curl -fsSL -o get_helm.sh https://raw.githubusercontent.com/helm/helm/master/scripts/get-helm-3
-chmod 700 $ORIGINAL_DIR/get_helm.sh
-$ORIGINAL_DIR/get_helm.sh
-
-if [[ ! $(command -v gsutil) ]]; then
-  CURRENT_DIR=$(dirname "$BASH_SOURCE")
-  . "${CURRENT_DIR}"/install-google-cloud-sdk.sh
-fi
+echo "Environment:"
+printenv
 
 export GOOGLE_APPLICATION_CREDENTIALS
 gcloud auth activate-service-account --key-file ${GOOGLE_APPLICATION_CREDENTIALS}
+gcloud -q auth configure-docker
 
 gcloud config set project ${GCLOUD_PROJECT}
 gcloud config set compute/region ${GCLOUD_REGION}
 gcloud config list
 
-echo "
+apt-get -qq update
+apt-get -y install wget build-essential gettext-base curl
+
+curl -fsSL -o get_helm.sh https://raw.githubusercontent.com/helm/helm/master/scripts/get-helm-3
+chmod 700 $ORIGINAL_DIR/get_helm.sh
+$ORIGINAL_DIR/get_helm.sh
+
+
+function getPublicAddresses() {
+  existing_addresses=$(gcloud compute addresses list --filter="region:($GCLOUD_REGION) name:kafka" --format "list(name)")
+  if [[ -z "$existing_addresses" ]]; then
+    echo "
 ============================================================
-Creating temp BQ table for Feast Serving
+Reserving IP addresses for Feast dependencies
 ============================================================
 "
-DATASET_NAME=feast_e2e_$(date +%s)
 
-bq --location=US --project_id=${GCLOUD_PROJECT} mk \
-  --dataset \
-  --default_table_expiration 86400 \
-  ${GCLOUD_PROJECT}:${DATASET_NAME}
+    gcloud compute addresses create \
```

Comment:
What happens if these IPs already exist? For example if multiple tests run at the same time.